plugins/lookup/hashi_vault.py

```diff
@@ -448,6 +453,12 @@ def __init__(self, **kwargs):
         if self.options['auth_method'] == 'token':
             client_args['token'] = self.options.get('token')
 
+        if self.options.get('proxy'):
+            client_args['proxies'] = {
+                'http': self.options['proxy'],
+                'https': self.options['proxy']
+            }
```

Comment:
@briantist I do not see why it's best to override only one scheme. Worse, it may fail when there is a redirection:

* url is in http://
* we use the proxy only for http
* a call to url return a redirection http→https
* the proxy isn't used → fail

Why do you want to only override one scheme ? Is it just a matter of taste… which I can understand :) or is there a argument I didn't think of.

Reply:
> Why do you want to only override one scheme ?

The main reason was just to set only what is needed, and not overstep.

The redirection scenario is a good example, though by the same logic that scenario could fail if we set both, as it's not guaranteed that the same proxy address should be used for both schemes.

Probably the most "correct" option is to rename this param `proxies` and take in the dictionary so that it can be fully user-defined. This would work, but it would no longer be possible to pass it through the term string, nor the single env var or ini entry. It would also be optimizing for the unusual case rather than the usual, which is why I didn't suggest it earlier.

Another option is to offer both params.. `proxy` being a single string that can be used to set one or both schemes, `proxies` being available as a true dict. Yet another option, is to independently define `proxy_http` and `proxy_https`... more options further complicates testing...

Something I'm trying to consider is that while this is a single lookup plugin now, I'm looking toward centralizing all of these connection options so that they can become shared by more plugins and even modules (the latter of which can have dict params passed in much more easily), and it's important that these things work consistently.

In the end, if we go with your suggestion we should document the behavior and note that it cannot be used with independent http/https proxies; as far as I know there'd be no fallback other than defining it via the env vars and setting it for the whole run (the thing you are trying to fix in this PR).

Let me think on this a bit more, in the meantime let me know if you have additional thoughts about the above.

Reply:
I think defining `proxy_http` and `proxy_https` is the simplest and most effective solution. I thought about it when I first wrote this patch, but I didn't need it for my particular use case so I went for `proxy`.
I will make the changes.

Reply:
would recommend holding off on the changes, two options means additional tests, etc. and requires duplication of the value if  you need it for both (which isn't the worst). Going to give it some more thought.

Reply:
done

Reply:
I am resolving this conversation, but if this solution doesn't suit you feel free to comment.